File: schemas/media-streaming/drizzle/playlist_followers.ts
// playlist_followers: Users who follow a playlist for updates.
// See README.md for full design rationale.

import { pgTable, uuid, timestamp, unique, index } from "drizzle-orm/pg-core";
import { playlists } from "./playlists";
import { users } from "./users";

export const playlistFollowers = pgTable(
  "playlist_followers",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    playlistId: uuid("playlist_id").notNull().references(() => playlists.id, { onDelete: "cascade" }),
    userId: uuid("user_id").notNull().references(() => users.id, { onDelete: "cascade" }),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    unique("uq_playlist_followers_playlist_id_user_id").on(table.playlistId, table.userId),
    index("idx_playlist_followers_user_id_created_at").on(table.userId, table.createdAt),
  ]
);

File: schemas/media-streaming/spacetimedb/playlist_followers.rs
// playlist_followers: Users following playlists for updates and recommendations.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = playlist_followers, public)]
pub struct PlaylistFollower {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub playlist_id: String, // UUID — FK → playlists.id (cascade delete)
    #[index(btree)]
    pub user_id: String, // UUID — FK → users.id (cascade delete)
    pub created_at: Timestamp,
}
// Composite unique: (playlist_id, user_id) — enforce in reducer logic
// Composite index: (user_id, created_at) — not supported; btree on user_id covers leading column.
